src/foam/nanos/controller/ApplicationController.js

```diff
@@ -395,6 +397,41 @@ foam.CLASS({
       });
     },
 
+    async function fetchLanguage() {
+      try {
+        let ctx = this.__subContext__;
+        let d = await  this.__subContext__.LocaleDAO;
+        d.select().then(e => {
+          console.log( e )
+          let arr = e.array;
+          arr.forEach(ea =>
+            {
+              let s = null;
+              try {
+                let i, obj;
+                do {
+                  i = ea.source.indexOf('.',++i);
+                  if (i != -1) {
+                    s = eval(ea.source.substring(0,i))
+                  }
+                } while (i > 0 && !!s);
+              }
+              catch(err) {
+                console.log(ea.source)
+                console.log(err)
+              }
+              if (!!s) {
+                s[ea.source.substring(ea.source.lastIndexOf('.')+1)] = ea.target;
+              }
+            })
+        })
+      } catch (err) {//TODO
+        this.notify(this.LANGUAGE_FETCH_ERR, 'error');
```

Comment:
@hchoura can you create a ticket to capture this scenario - we need to 'notify' ourselves - send the error back to our noc that this situation is occurring.